### PR TITLE
issue #151

### PR DESCRIPTION
**Pull request by vincentneko**

### CODE REVIEW

File: src/Game.ts [hunk 1]
```diff
@@ -383,6 +382,7 @@ export class Game {
       if (this.gameIsOver()) {
         this.gotoFinalGreeneryPlacement();
       } else {
+        this.generation++;
```
Comment (alrusdi) [inline src/Game.ts]: I think generation must be incremented before
if (this.gameIsOver()) {
this leads to that bug with never ending solo game

**Comment by vincentneko** [comment]
> Drafting code needs more tests. @vincentneko can you please add one at least for this issue

Yes you are right, I will try to put something up.

File: src/Game.ts (hunk 1 above)
Comment (vincentneko) [reply src/Game.ts]: Not sure about this.
I've made the change I want to push in this PR and my solo game went to generation 12, you can see it in the screenshot I've sent.

And also, before implementing drafting option, I think generation was incremented in gotoResearchPhase.

**Comment by vincentneko** [comment]
> > Drafting code needs more tests. @vincentneko can you please add one at least for this issue
> 
> Yes you are right, I will try to put something up.

I've tested generation increment for solo and 2 players (draft and no draft options)
Not sure how I can test drafting by itself though. Is there a way to answer the SelectCard function the player has to choose when in drafting ?

File: src/Game.ts (hunk 1 above)
Comment (alrusdi) [reply src/Game.ts]: I think it is good start. We will expand coverage someday.
You can set any answer to Input function by calling its cb function (CallBack). See example in Plantation.spec.ts

**Pull request merged**